### PR TITLE
fix: sanitize Content-Disposition filename in download URL

### LINKED ISSUE #55: fix(backend): sanitize Content-Disposition filename in download URL
## Problem
`StorageService.generate_download_url()` builds `ResponseContentDisposition` by interpolating a user-controlled filename directly:

```python
params["ResponseContentDisposition"] = f'attachment; filename="{filename}"'
```

The filename originates from:
- `user_resource.name` (user-editable via `PATCH /api/v1/resources/{id}`)
- `resource.file_name` (user-controlled at upload)

This path is used by `GET /api/v1/resources/{resource_id}/download`.

## Why this is dangerous
This is an HTTP header injection sink (`Content-Disposition`) with unsanitized user input.

Crafted filenames (CR/LF, quotes, semicolons, control chars, parser-confusing bytes) can:
- break header structure / cause response-splitting behavior
- inject or corrupt downstream headers in weak proxy/CDN/client paths
- create content-type or cache interpretation inconsistencies
- produce inconsistent browser behavior when parsing `Content-Disposition`

Even if one layer blocks some payloads, this remains high risk because behavior differs across SDK/runtime/CDN/proxy/browser combinations.

## Expected fix
- Sanitize/normalize filename before putting it in header params.
- Strip/reject control chars (especially CR/LF).
- Escape or avoid raw quoted interpolation.
- Emit standards-compliant header values (e.g., safe ASCII fallback + RFC 5987 `filename*`).
- Add tests for malicious filename payloads.

## Affected code
- `backend/app/services/storage_service.py` (`generate_download_url`)
- `backend/app/api/v1/resources.py` (`GET /{resource_id}/download` source of filename)

## Severity
Security issue (header injection / response-splitting class).

### PR DESCRIPTION
**Pull request by hsilabot**
## Summary
Fixes unsafe `Content-Disposition` filename construction for resource download URLs by sanitizing user-controlled filenames before they are passed into presigned R2 response headers.

Closes #55.

## Root cause
- `generate_download_url()` built `ResponseContentDisposition` by directly interpolating `filename`.
- `filename` is user-controlled via:
  - `user_resource.name` (rename flow)
  - `resource.file_name` (upload metadata)
- This allowed header-context injection risk (e.g., control chars, CR/LF, parser-confusing punctuation).

## Security impact
Without sanitization, crafted filenames can cause header parsing ambiguity and, in weak downstream chains, open response-splitting/header-injection style behavior.

### Backend
- Updated `backend/app/services/storage_service.py`:
  - Added safe filename normalization for `Content-Disposition` usage.
  - Strips/replaces control characters and header-significant punctuation.
  - Produces a safe ASCII filename value.
  - Falls back to a safe default (`download`) when sanitized value is empty.

### Tests
- Added/updated targeted tests in `backend/tests/test_storage_service.py` for:
  - normal filenames
  - CR/LF and control-char payloads
  - quote/semicolon-style injection attempts
  - non-ASCII filename behavior

## Verification
Executed:

```bash
./.venv/bin/python -m pytest tests/test_storage_service.py tests/test_settings.py
```

Result: `9 passed`.

## Risk / compatibility
- Low runtime risk, scoped to download header construction.
- Expected user-visible behavior remains the same for normal filenames.
- Edge-case filenames are normalized to safe output.

### CODE REVIEW

**Comment by HSILA** [comment]
@hsilabot spawn a gpt54 to review this. add an actual review here in the PR.

File: backend/app/services/storage_service.py [hunk 1]
```diff
@@ -39,6 +46,16 @@ def __init__(self, settings: Settings) -> None:
...
         return self._bucket_name
 
+    @staticmethod
+    def _sanitize_download_filename(filename: str) -> str:
+        normalized = unicodedata.normalize("NFKD", filename)
+        ascii_filename = normalized.encode("ascii", "ignore").decode("ascii")
+        without_controls = _CONTROL_CHARS_RE.sub("_", ascii_filename)
+        sanitized = _FILENAME_UNSAFE_CHARS_RE.sub("_", without_controls)
+        sanitized = _FILENAME_REPEAT_UNDERSCORE_RE.sub("_", sanitized)
+        sanitized = _FILENAME_REPEAT_SPACE_RE.sub(" ", sanitized).strip(" ._")
```
Comment (hsilabot) [inline backend/app/services/storage_service.py]: [P2] This strips leading/trailing dots from the entire sanitized filename, which means filenames whose basename is entirely removed by ASCII-folding/cleanup can lose their extension separator. For example, `"文件.pdf"` and `"😈.pdf"` currently become `"pdf"` instead of something like `"download.pdf"` or `"file.pdf"`. That is a user-visible regression for legitimate non-ASCII names, and the current tests do not cover it. Please preserve the extension when the stem sanitizes away and add a regression test for a non-ASCII-only basename.

**Review by hsilabot — COMMENTED** [review]
Left one medium-priority edge-case note on filename sanitization; otherwise the header-injection fix looks sound.

**Comment by HSILA** [comment]
@hsilabot spawn a gpt54 to review this PR with inline comments.
